### PR TITLE
Render programmatic text decoration

### LINKED ISSUE #278: Text-Decoration is not rendered
I create a simple SvgDocument with a SvgText element. For the SvgText element the property TextDecoration is set to SvgTextDecoration.LineThrough.
If I export it as XML the XML contains the attribute text-decoration="line-through", but the text is rendered without the "line-through" when I save it as PNG.

Example is attached. In a Web-Browser it is rendered correctly.

Image

Image

### PR DESCRIPTION
**Pull request by wieslawsoltes**
# PR Summary

Fixes #278.

## Summary

This change makes programmatically assigned `SvgText.TextDecoration` values render in the Skia scene graph text pipeline. Previously, parsed SVG markup with a raw `text-decoration` attribute could flow through the renderer, but documents created in code could set `TextDecoration = SvgTextDecoration.LineThrough`, serialize correctly as `text-decoration="line-through"`, and still render without the decoration in PNG output.

The renderer now checks the normal `text-decoration` attribute value when the parser-specific raw text-decoration custom attribute is absent. This preserves the existing raw SVG parsing behavior for authored SVG values while allowing enum-backed programmatic values such as `LineThrough` to participate in the same decoration drawing pipeline.

## Implementation Details

- Added a named `text-decoration` attribute constant in `SvgSceneTextCompiler`.
- Kept raw parsed SVG text-decoration handling as the first path, so existing markup parsing behavior remains unchanged.
- Added a fallback for programmatic attribute values exposed through `TryGetAttribute("text-decoration", ...)`.
- Reused the existing SVG token parser for SVG string values such as `line-through`.
- Added enum-name parsing for programmatic values such as `LineThrough`.
- Rejected numeric enum strings in the fallback parser so arbitrary numeric values are not treated as valid SVG text-decoration input.

## Test Coverage

Added an end-to-end PNG rendering regression test:

- Builds a `SvgDocument` in code.
- Creates one text element with `SvgTextDecoration.None`.
- Creates a second otherwise identical text element with `SvgTextDecoration.LineThrough`.
- Renders both documents through `SKSvg.FromSvgDocument(...).Save(...)`.
- Compares the resulting images and asserts that the line-through version changes the rendered pixels.

## Validation

Ran the focused affected test sets on `net10.0`:

```bash
dotnet test tests/Svg.Skia.UnitTests/Svg.Skia.UnitTests.csproj -f net10.0 -c Release --no-restore --filter "FullyQualifiedName~SKSvgTests" --logger "console;verbosity=minimal"
dotnet test tests/Svg.Skia.UnitTests/Svg.Skia.UnitTests.csproj -f net10.0 -c Release --no-restore --filter "FullyQualifiedName~SvgSceneTextCompilerTests" --logger "console;verbosity=minimal"
```

Results:

- `SKSvgTests`: 8 passed, 0 failed.
- `SvgSceneTextCompilerTests`: 19 passed, 0 failed.

## Notes

The full solution build with `--no-restore` was not usable in this fresh worktree before committing because many `project.assets.json` files were missing, and the local environment also lacks the `net461` reference assemblies required by the full solution target graph. The focused `net10.0` tests above were restored and then rerun with `--no-restore`.

### CODE REVIEW

File: src/Svg.SceneGraph/SvgSceneTextCompiler.cs [hunk 1]
```diff
@@ -3984,8 +3985,15 @@ private static SvgTextBase ResolveDecorationPaintSource(SvgTextBase leafTextBase
     private static bool TryGetOwnTextDecoration(SvgElement element, out SvgTextDecoration decorations)
     {
         decorations = SvgTextDecoration.None;
-        return element.CustomAttributes.TryGetValue(SvgStyleAttributeNames.RawTextDecorationAttributeKey, out var rawValue) &&
-               TryParseTextDecorationValue(rawValue, out decorations) &&
+        if (element.CustomAttributes.TryGetValue(SvgStyleAttributeNames.RawTextDecorationAttributeKey, out var rawValue))
+        {
+            return TryParseTextDecorationValue(rawValue, out decorations) &&
+                   HasRenderableDecorations(decorations);
```
[the remark this answers is not in the document]
Comment (wieslawsoltes) [reply src/Svg.SceneGraph/SvgSceneTextCompiler.cs:3991]: Addressed in `f584514ce2f35838f8ad1b6e2ed0d7519d9b98d8`.

The fix keeps the raw `text-decoration` cache authoritative for freshly parsed SVG, so invalid authored values such as comma-separated decoration lists are still rejected. The stale-cache problem is handled at the local attribute collection layer instead: when an existing `text-decoration` attribute is set again, the cached `__svgskia:text-decoration-raw` custom attribute is removed before the change notification. That matches the programmatic `SvgText.TextDecoration` setter path without changing external submodules.

I added regression tests for both cases called out here:
- parsed `text-decoration="none"` followed by `TextDecoration = SvgTextDecoration.LineThrough`
- parsed `text-decoration="line-through"` followed by `TextDecoration = SvgTextDecoration.None`

I also reran the comma-separated resvg text-decoration row that caught the raw parsing edge case.

Validation:
- `dotnet format Svg.Skia.slnx --no-restore --include src tests samples`
- `dotnet test tests/Svg.Skia.UnitTests/Svg.Skia.UnitTests.csproj -f net10.0 -c Release --no-restore --filter "FullyQualifiedName~SKSvgTests.Save_ProgrammaticTextDecorationLineThrough_RendersDecoration|FullyQualifiedName~SKSvgTests.Save_ParsedTextDecoration|FullyQualifiedName~resvgTests.a_text_decoration"`
- `dotnet build Svg.Skia.slnx -c Release`
- `dotnet test Svg.Skia.slnx -c Release`
